packages/react/src/primitives/Checkbox/Checkbox.tsx

```diff
@@ -7,24 +8,27 @@ import { Input } from '../Input';
 import { Text } from '../Text';
 import { VisuallyHidden } from '../VisuallyHidden';
 import { CheckboxProps } from '../types/checkbox';
-import { Primitive } from '../types/view';
+import { PrimitiveWithForwardRef } from '../types/view';
 import { splitPrimitiveProps } from '../shared/styleUtils';
 import { ComponentClassNames } from '../shared/constants';
 import { useTestId } from '../utils/testUtils';
 
-export const Checkbox: Primitive<CheckboxProps, 'input'> = ({
-  checked,
-  className,
-  defaultChecked,
-  hasError,
-  isDisabled,
-  label,
-  labelHidden,
-  onChange: onChangeProp,
-  size,
-  testId,
-  ..._rest
-}) => {
+const CheckboxInner: PrimitiveWithForwardRef<CheckboxProps, 'input'> = (
```

Comment:
non-related: if Checkbox is only being used internally, should be a Primitive? Maybe should be a simple React.FC component instead.

Reply:
We still want to expose the intrinsic attributes from the underlying DOM element, so keeping it as Primitive seems better.